includes/promoted-jobs/class-wp-job-manager-promoted-jobs-api.php

```diff
@@ -120,6 +144,31 @@ public function prepare_item_for_response( $item, $request ) {
 		];
 		return rest_ensure_response( $data );
 	}
+
+	/**
+	 * Update the promoted job status.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function update_job_status( $request ) {
+		$post_id = $request->get_param( 'id' );
+		$status  = $request->get_param( 'status' );
+		$post    = get_post( $post_id );
+
+		if ( empty( $post ) ) {
+			return new WP_Error( 'not_found', __( 'The promoted job was not found', 'wp-job-manager' ), [ 'status' => 404 ] );
+		}
+
+		$result = update_post_meta( $post_id, WP_Job_Manager_Promoted_Jobs::META_KEY, $status );
```

Comment:
> My only question is - do we care if it's a 0 or 1? Right now there's no validation to make sure it's boolean but just that it's numeric, so technically it could be _promoted => 29847592876542987549082745 and it would still work.

Good point here! Actually, the feed endpoint is getting the posts with `_promoted` as `"1"`. So maybe it should  be something like this?

```php
$result = update_post_meta( $post_id, WP_Job_Manager_Promoted_Jobs::META_KEY, $status ? '1' : '0' );
```

Reply:
Just FYI, I fixed this on #2488. :)